adapter/adapter-protocol/src/codelldb.rs

```diff
@@ -268,6 +268,7 @@ pub struct AdapterSettings {
     pub evaluate_for_hovers: Option<bool>,
     pub command_completions: Option<bool>,
     pub reproducer: Option<Either<bool, String>>,
+    pub enable_terminate_request: Option<bool>,
```

Comment:
these adapter settings are a "custom" message invented by codelldb and only usable in vscode.

Can we avoid adding more? It means that this option is not available in other DAP-compatible clients.

If this behaviour is useful and correct, then why would we not always offer it to all clients?

Reply:
I believe this entire MR (and the underlying reported issue) are only applicable to vscode.  This new optional shutdown behavior was (somewhat) recently added in vscode.  I'm unfamiliar with DAP - does it also have a query for whether a terminate request is supported?

Reply:
It's a standard request supported by many adapters: https://microsoft.github.io/debug-adapter-protocol/specification#Requests_Terminate

Reply:
My point is that if the server supports the terminate request, it should say it does, then it's up to the client to decide whether or not to use it _globally_ not in a spacial way just for one adapter. Users are unlikely to change this setting, nor to really understand what chaining it means. The behviour of the client is consistent among adapters supporting the request, without additional configuration.

Reply:
Oh I see... so you're suggesting that this should just be implemented without a configuration item to toggle it?  That would certainly be easier, I just added the configuration item (and all the associated plumbing) because I thought this PR would be more likely to be accepted if it didn't change the user experience without opting in.  I think it would only be noticed if the debuggee implements a SIGINT handler, in which case it's probably the desired behavior anyways.  I'll change that (probably in the morning).

Reply:
Well, ultimately it's a question for @vadimcn to decide - I'm just expressing my preference as a client maintainer.

Reply:
Just pushed an update that removes the configuration and enables the behavior on unix platforms.